### ISSUE TITLE
Icon replaced by white rectangle

### ISSUE THREAD

**Issue by mathematicalcoffee** [mathematicalcoffee/EvilStatusIconForever#1]
Sometimes the tray icons get replaced by white rectangles. Observed in GNOME 3.6 only.

There's a report of this happening after fullscreening a video (I have not tested personally yet but have noticed the white rectangles every now and then).

**Comment by mathematicalcoffee** [comment]
I can confirm this happens whenever you fullscreen an application.

Now, what's the problem?

All I can think of is that it is to do with the Shell.TrayIcon, but I can't touch that.

What else could it be? The Shell.TrayIcon is nested inside a PanelMenu.Button.actor.

**Comment by dziq** [comment]
This problem still exists. Eg. when I try to use thunderbird with minimize to tray extension.

**Comment by mathematicalcoffee** [comment]
Can you give more details?

Specifically, when you fullscreen and un-fullscreen a window, do you have a white square?

If not, exactly what causes the white square (this will need to be opened under a new bug)?

What version of gnome shell do you have, what's the minimize to tray extension so I can install it (I can't find it on e.g.o), and what **exact** steps must I do to get this bug to occur (e.g. start thunderbird, minimize, unminimize, icon is gone, or something??)

**Comment by dziq** [comment]
Ok then. I've checked it on Archlinux and Fedora 18 both with gnome-shell 3.6.2. It looks like this http://wstaw.org/m/2012/12/05/gnome-icon.png

It happens when I want to click on thunderbird tray icon (this extension: https://addons.mozilla.org/en-US/thunderbird/addon/minimizetotray-revived/?src=search) and want to have thunderbird window open.
But it also happend when I try to close Dropbox from tray icon. Same white sqare.

If you want to reproduce just do as you mentioned  start thunderbird, minimize, unminimize and there should be this square.
Regards
BN

**Comment by varemenos** [comment]
Why is this closed? I can confirm that this bug still exists.

**Comment by mathematicalcoffee** [comment]
Are you using the version from this repository (clone from master)? I haven't submitted a pull request for this fix to the original repo so none of the e.g.o versions or the original repo will have the fix.

Also I sometimes get the white square too but not as a result of fullscreen/un-fullscreen - I haven't worked out how to reproduce it yet. If you can reliably get the white square please let me know how.

For GNOME 3.6 I have found that [TopIcon](https://extensions.gnome.org/extension/495/topicons/) works quite nicely instead of this and seems not to have the white square problem.

**Comment by varemenos** [comment]
Yes im using your repository not the "original".
Here is a way i found to reproduce it successfully every time:
On startup pidgin is minimized on tray by default (even though it doesn't appear in the topbar but in the notification bar)(its included in the list with the application to appear in the topbar by default and i haven't removed it from the list). If i launch pidgin while its closed to tray, another white square will appear.

I've already tried topIcon but I don't like cause it doesn't allow me to choose which ones i want to put in the topbar.

**Comment by PaterSiul** [comment]
I can confirm it, too.
In my case the issue shows up when the program quits or is terminated.

I'm running Arch with gnome-shell 3.6.2 and this repository

**Comment by mathematicalcoffee** [comment]
Yeah I think I know what this is, I think I forgot to listen to the program closing (e.g. closing dropbox) to remove the icon.
